tmt/steps/execute/__init__.py

```diff
@@ -562,31 +562,65 @@ def load_tmt_report_results(self, invocation: TestInvocation) -> list["tmt.Resul
             return []
 
         # Check the test result
-        self.debug(f"tmt-report-results file '{report_result_path} detected.")
+        self.debug(f"tmt-report-results file '{report_result_path}' detected.")
+
+        # The worst result outcome we can find among loaded results...
+        original_outcome: Optional[ResultOutcome] = None
+        # ... and the actual outcome we decided is the best representing
+        # the results.
+        # The original one may be left unset - malformed results file,
+        # for example, provides no usable original outcome.
+        actual_outcome: ResultOutcome
+        note: Optional[str] = None
+
+        results = tmt.utils.yaml_to_list(self.read(report_result_path))
 
-        with open(report_result_path) as result_file:
-            result_list = [line for line in result_file.readlines() if "TESTRESULT" in line]
-        if not result_list:
+        if not results:
             raise tmt.utils.ExecuteError(
                 f"Test result not found in result file '{report_result_path}'.")
-        result = result_list[0].split("=")[1].strip()
-
-        # Map the restraint result to the corresponding tmt value
-        actual_result = ResultOutcome.ERROR
-        note: Optional[str] = None
 
         try:
-            actual_result = ResultOutcome(result.lower())
-        except ValueError:
-            if result == 'SKIP':
-                actual_result = ResultOutcome.INFO
-            else:
-                note = f"invalid test result '{result}' in result file"
+            outcomes = [
+                ResultOutcome.from_spec(result.get('result'))
+                for result in results
+                ]
+
+        except tmt.utils.SpecificationError as exc:
+            actual_outcome = ResultOutcome.ERROR
+            note = exc.message
+
+        else:
+            hierarchy = [
+                ResultOutcome.SKIP,
+                ResultOutcome.PASS,
+                ResultOutcome.WARN,
+                ResultOutcome.FAIL]
+            outcome_indices = [hierarchy.index(outcome) for outcome in outcomes]
+            actual_outcome = original_outcome = hierarchy[max(outcome_indices)]
```

Comment:
Do you think it would make sense to add magic methods for comparison to ResultOutcome?
You could then directly do `max(outcomes)`
I know this can get difficult and surely won't lower number of lines, but IMO more places could utilise it
@happz what do you think? is it reasonable? or am I just asking for unnecessary overengineering

Reply:
I'd say it would work, although I wouldn't block on that. But yes, other places might benefit from results being sortable.

Reply:
I attempted to implement this, but couldn't get it working so I've left it undone.